File: Java/프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    private int getGCD(int a, int b) {
        return b == 0 ? a : getGCD(b, a % b);
    }
    
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int numer = numer1 * denom2 + numer2 * denom1;
        int denom = denom1 * denom2;
        
        int gcd = getGCD(numer, denom);
        
        int[] answer = {numer / gcd, denom / gcd};
        return answer;
    }
}

File: Kotlin/프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.kt
class Solution {
    fun gcdRecursive(a: Int, b: Int): Int {
        return if (b == 0) a else gcdRecursive(b, a % b)
    }
    
    fun solution(numer1: Int, denom1: Int, numer2: Int, denom2: Int): IntArray {
        
        var numer = numer1 * denom2 + numer2 * denom1
        var denom = denom1 * denom2
        
        val gcd = gcdRecursive(numer, denom)
        
        numer /= gcd
        denom /= gcd
        
        val answer = intArrayOf(numer, denom)
        return answer
    }
}